### DISCUSSION TITLE
DICOM Commercial Extension Built with the Mirth Connect Development Team & Community

### DISCUSSION THREAD

**Discussion by twest-mirthconnect** [nextgenhealthcare/connect#5549] [General]
Hello DICOM engineers! 

Would like to add your name to a great new extension for Mirth Connect? Would you like to help your organization use this powerful new extension in perpetuity? If this sounds like something you would like to work on with 20+ year subject matter experts already working on the framework, reach out to Travis West (twest@nextgen.com) and/or Emily McDonald (emcdonald@nextgen.com).

**Comment by tom08zehn** [comment]
Sounds interesting. Can you tell us more about the DICOM features it supports?

**Reply by twest-mirthconnect** [reply]
Our goal is to expand into the following areas in stages (details to be determined):

- Support for the newest version of the toolkit (DCM4CHE 2.x & 5.x)
- New functionalities
- Add Support/GUI location to add additional SOP classes supported by SCU/SCP
- Support/Create Temp Db Tables For C-STORE, C-MOVE in transit
- Embed a viewer and temp storage
- Workstation capabilities
- Documents To DICOM + C-STORE to 1+ Destinations After Conversion
- Add Support for ImageMagick
- DICOM Modality Worklist provider	
HL7 Mapping
DICOM Mapping

**Reply by pacmano1** [reply]
What is the expected cost for this new extension?    And do you have a competitive analysis?

**Comment by twest-mirthconnect** [comment]
TBD on the price

Yes, competitive analysis is based on SMEs working on the project and direct market research conducted over the last year or so

**Reply by pacmano1** [reply]
I meant can you provide the competitive analysys here so others can see how Mirth will stack up.   Understood on price.

**Comment by twest-mirthconnect** [comment]
At some point in the future, sure.

**Comment by MuhammadAlibutt** [comment]
hey guys I NEED SOME HELP. 
I am transferring JSON data from one Channel to other and its working fine, as data is being transferred. on the Destination Channel, 
i am trying to convert the coming data to HL7V2 form
following is the code I wrote for conversion:
var json = connectorMessage.getEncodedData();

var arr = JSON.parse(json);

var msg = <HL7Message/>;
var tmp ;
msg=tmp;
createSegment("MSH",msg);
msg['MSH']['MSH.1']= tmp['MSH'][0]['Segament'];
msg['MSH']['MSH.2']= tmp['MSH'][0]['Encoding_Characters'];
msg['MSH']['MSH.3']['MSH.3.1']=tmp['MSH'][0]['Sending_Application_Name'];
var message = SerializerFactory.getSerializer('HL7V2').fromXML(msg);
logger.debug(message);


following error i am getting 
TypeError: Cannot read property "MSH" from undefined

**Reply by pacmano1** [reply]
@MuhammadAlibutt , please do not hijack threads, your post has nothing to do with the initial post.

Also please cross reference your posts if you post to the forums and github (for other see https://forums.mirthproject.io/forum/mirth-connect/support/7921-hl7-to-json-objects#post182821)

**Comment by tanqan** [comment]
Hi, are there news about this new extension?

**Comment by jmiddleton** [comment]
Any updates on thi? I would like to help

**Reply by jonbartels** [reply]
The NextGen staff behind this initiative (Travis and Emily) have not worked at NextGen in over one year. I assume this initiative is no longer active. :/
